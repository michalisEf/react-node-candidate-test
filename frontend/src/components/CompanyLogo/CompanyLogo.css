.company-logo {}

.company-logo img {
  width: 5rem
}