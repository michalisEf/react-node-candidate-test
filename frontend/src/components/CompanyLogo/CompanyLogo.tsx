import React from 'react';

import { ICompanyLogo } from '../../interfaces';

import './CompanyLogo.css';

const CompanyLogo = ({ company, logo }: ICompanyLogo): JSX.Element => {
  return (
    <div className="company-logo">
      <img src={logo} alt={company} />
    </div>
  );
};

export default CompanyLogo;
