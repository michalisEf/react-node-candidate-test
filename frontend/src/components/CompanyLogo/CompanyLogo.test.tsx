import React from 'react';
import { render, screen } from '@testing-library/react'

import CompanyLogo from './CompanyLogo';

it('renders CompanyLogo component', () => {
	const props = {
		logo: "http://host.com/photo.jpg",
		company: "acme"
	};
	render(<CompanyLogo {...props} />);
	const image = screen.getByRole('img');
	expect(image).toHaveAttribute('src', 'http://host.com/photo.jpg');
	expect(image).toHaveAttribute('alt', 'acme');
	expect(image).toMatchSnapshot();

});
