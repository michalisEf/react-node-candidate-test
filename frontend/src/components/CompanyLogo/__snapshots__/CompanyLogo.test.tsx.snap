// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CompanyLogo component 1`] = `
<img
  alt="acme"
  src="http://host.com/photo.jpg"
/>
`;
